src/python/modules/ensembl/compara/citest/tabletest.py

```diff
@@ -0,0 +1,245 @@
+
+from sqlalchemy import create_engine
+from sqlalchemy.sql import text
+from ensembl.compara.citest.test import Test
+
+
+## This class extend the base class Test to test data into the database
+## this test start with an intialisation
+##
+class TableTest(Test):
+
+    def __init__(self):
+        Test.__init__(self)
+        self.ref_connection_engine=None
+        self.target_connection_engine=None
+        self.table_tests=[]
+
+
+    def initialise_tests(self,dic_argument): # str_pipeline_name, obj_ref_connection, obj_target_connection, lst_table_tests):
+        self.pipeline_name=dic_argument["str_pipeline_name"]
+        self.ref_connection_engine=dic_argument["obj_ref_connection_engine"]
+        self.target_connection_engine=dic_argument["obj_target_connection_engine"]
+        self.table_tests=dic_argument["lst_table_tests"]
+
+        ###Test that the all tables define in lst_table_tests exists in the databases
+        lst_tables_database=self._get_pipeline_table(self.ref_connection_engine)
+        for dic_tabl_test in self.table_tests:
+            str_table_name=dic_tabl_test["name"]
+            if not str_table_name in lst_tables_database:
+                raise Exception("Error: table name '"+str_table_name+"' is not in the database")
+
+
+    def _get_pipeline_table(self, obj_connection_engine):
+            TABL_NAME=0
+            lst_pipeline_table_names=[]
+            str_request="SHOW TABLE STATUS WHERE Rows > 0;"
+            obj_request = text(str_request)
+            obj_connection=obj_connection_engine.connect()
+            obj_result=obj_connection.execute(obj_request)
+            for row in obj_result:
+                str_table_name=row[TABL_NAME]
+                lst_pipeline_table_names.append(str_table_name)
+            obj_connection.close()
+            return lst_pipeline_table_names
+
+
+    def run_tests(self):
+        for dic_tabl_test in self.table_tests:
+            str_table_name=dic_tabl_test["name"]
+            int_line_variation=dic_tabl_test["variation"]
+            lst_group_by=dic_tabl_test["lst_group_by"]
+            lst_columns=dic_tabl_test["columns"]
+
+            #First it compare the number of row between the ref and target db
+            self._test_table_number_row(str_table_name, int_line_variation)
```

Comment:
This information could easily be fetched when fetching the list of table names from the `INFORMATION_SCHEMA.TABLES` table above, reducing the number of queries we would need